### PR TITLE
Fix config structure name in pragma for SeparableConv1D

### PR DESCRIPTION
**Pull request by qberthet**
# Description

Simple fix for #883

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Tested with example code provided in bug report #883 : Fail before applying this commit, works afterward.

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by jmitrevs** [comment]
Is it possible to add a pytest to show that there is a problem and that this fixes it? We don't do synthesis in pytests but we do compile the HLS code.

**Comment by qberthet** [comment]
I'll try to add this later today.

File: hls4ml/templates/vivado/nnet_utils/nnet_sepconv_stream.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ void depthwise_product(data_T data[CONFIG_T::kernel_size * CONFIG_T::n_chan], re
 
     #pragma HLS ARRAY_PARTITION variable=mult complete
 
-    #pragma HLS ALLOCATION operation instances=mul limit=CONFIG_T::multiplier_limit
+    #pragma HLS ALLOCATION operation instances=mul limit=CONFIG_T::mult_config::multiplier_limit
```
Comment (vloncar) [inline hls4ml/templates/vivado/nnet_utils/nnet_sepconv_stream.h:26]: This technically works, but we deliberately didn't use the `mult_config` in `depthwise_product`. In 2D the config has the `multiplier_limit` and we should use the same approach. So below [this ](https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/vivado/passes/convolution_templates.py#L43) line we should add to [these](https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/vivado/passes/convolution_templates.py#L146-L147) lines.

**Review by vloncar — CHANGES_REQUESTED** [review]

File: hls4ml/templates/vivado/nnet_utils/nnet_sepconv_stream.h (hunk 1 above)
Comment (qberthet) [reply hls4ml/templates/vivado/nnet_utils/nnet_sepconv_stream.h:26]: Ok, I was suspecting that there was something else to it, but not familiar with the code base yet. I'll start by adding a pytest that trig the issue, add the fix you suggest and force push to this PR.

**Comment by qberthet** [comment]
@jmitrevs : I've added a test, but the problem only appear during CSynth, not CSim, so not sure if that will be ok (requires vivado). Also had to catch the return code of vivado_hls command to raise exception and make the test fail. Let me know if you prefer that I remove the two first commit to only have the fix.

@vloncar : your fix obviously does the job. Thanks!

**Comment by qberthet** [comment]
And I just realize that my test is not setting the output folder correctly. Let me know if you want to keep these tests before I fix it.

**Review by vloncar — APPROVED** [review]

**Pull request merged**